internal/component/discovery/target_builder.go

```diff
@@ -0,0 +1,197 @@
+package discovery
+
+import (
+	commonlabels "github.com/prometheus/common/model"
+
+	"github.com/grafana/alloy/internal/component/common/relabel"
+)
+
+type TargetBuilder interface {
+	relabel.LabelBuilder
+	Target() Target
+	SetKV(kv ...string) TargetBuilder
```

Comment:
SetKV name seems odd, since I think of key value but neither the input or exposed property is inherently a KV. It returns a TargetBuilder, `SetTargetBuilder`, `CopyTargetBuilder`?